### PR TITLE
Rename: Writer/Reader => Transmitter/Receiver.

### PR DESCRIPTION
**Pull request by htmldoug**
## Confusion: Reader/Writer
1. `weePickle.Reader`/`weePickle.Writer` deal heavily with `java.io.Reader`/`java.io.Writer`. Confusing.
2. You have `pony: Pony` and want a `JsValue`. Do you need a `weePickle.Reader[JsValue]` or a `weePickle.Writer[JsValue]`? How long did it take you to reach your answer? (spoilers: [answer](https://gist.github.com/htmldoug/a21bf6d87f8c3fd575d311dfb5dae713))

## Renaming
For these reasons, Reader/Writer need a rename--a non-trivial task.

`Transmitter`/`Receiver` is the best I can come up with. I guess this makes the Visitor a radio or modem, but Visitor is unambiguous, so I'm leaving that one alone.

## Usage
```scala
FromScala(JsArray(1,2,3)).transmit(ToScala[Seq[Int]]) ==> List(1, 2, 3)
```

This requires a `Transmitter[JsArray]` and `Receiver[Seq[Int]]`.

@coconutpalm @vito-c @spemmons @danny-goldberg-rally @praveenravi77 @AndyKirsch

### CODE REVIEW

File: bench/src-js/Main.scala [hunk 1]
```diff
@@ -36,7 +36,7 @@ object Main{
     )
   }
   def weepickleWebDefault(duration: Int) = {
-    import com.rallyhealth.weepickle.v1.WeePickle.{ReaderWriter => RW}
+    import com.rallyhealth.weepickle.v1.WeePickle.{Transceiver => RW}
```
Comment (vito-c) [inline bench/src-js/Main.scala]: OH and Transceivers do both!

**Comment by htmldoug** [comment]
@arkban, you're invited, too.

File: bench/src/com/rallyhealth/weepickle/v1/ClassDefs.scala [hunk 2]
```diff
@@ -1,7 +1,7 @@
 package com.rallyhealth.weepickle.v1
 
 //import com.rallyhealth.weepickle.v1.ADTs.ADT0
-import WeePickle.{ReaderWriter => RW, Reader => R, Writer => W}
+import WeePickle.{Transceiver => RW, Receiver => R, Transmitter => W}
```
Comment (vito-c) [inline bench/src/com/rallyhealth/weepickle/v1/ClassDefs.scala]: nitpick: `import WeePickle.{Transceiver => TX, Receiver => R, Transmitter => T}`

Comment (danny-goldberg-rally) [reply bench/src/com/rallyhealth/weepickle/v1/ClassDefs.scala]: TX? isn't that transmission?

**Pull request merged**